go/tools/builders/asm.go

```diff
@@ -94,6 +94,12 @@ func buildSymabisFile(goenv *env, sFiles, hFiles []fileInfo, asmhdr string) (str
 			seenHdrDirs[hdrDir] = true
 		}
 	}
+	// The package path has to be specified as of Go 1.22 or the resulting
+	// object will be unlinkable, but the -p flag is only required in
+	// preparing symabis since Go1.22
+	if packagePath != "" && isGoMinorVersionEqualHigher(22) {
```

Comment:
Since when has `go build` been passing the package path? I would slightly prefer mimicking its behavior, even if that means changing command lines with Go 1.21 and earlier. With our current CI setup, we otherwise wouldn't have coverage for this code path as we are still on 1.21.

Reply:
Since Go1.19: https://go-review.googlesource.com/c/go/+/404298

However, not passing package path finally broke in August 2023 on go1.22-devel: https://go-review.googlesource.com/c/go/+/523336/7/src/cmd/asm/internal/asm/parse.go

Reply:
But fair, I'll change the logic so that both branches are for Go1.19?